lib/java_buildpack/framework/sealights_agent.rb

```diff
@@ -0,0 +1,168 @@
+# frozen_string_literal: true
+
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2020 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'java_buildpack/component/versioned_dependency_component'
+require 'java_buildpack/framework'
+require 'uri'
+require 'net/http'
+require 'pathname'
+require 'base64'
+require 'json'
+
+module JavaBuildpack
+  module Framework
+
+    # Encapsulates the functionality for enabling zero-touch Sealights support.
+    class SealightsAgent < JavaBuildpack::Component::BaseComponent
```

Comment:
It is much preferred to use the VersionedDependencyComponent as a base. It will handle all of the download and caching logic for you, something that is not trivial to re-implement. 

https://github.com/cloudfoundry/java-buildpack/blob/main/docs/extending-versioned_dependency_component.md

Is there a specific reason you opted to implement base? Like did VersionedDependencyComponent not work for some reason?

Reply:
Yes, I had to set a token in order to download the agent and wasn't sure how to do that.